Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Storefront.m

```diff
@@ -75,19 +75,31 @@ - (BUYRequestOperation *)getProductsPage:(NSUInteger)page completion:(BUYDataPro
 	}];
 }
 
-- (BUYRequestOperation *)getProductById:(NSString *)productId completion:(BUYDataProductBlock)block;
+- (BUYRequestOperation *)getProductByHandle:(NSString *)handle completion:(BUYDataProductBlock)block
+{
+	BUYAssert(handle, @"Failed to get product by handle. Product handle must not be nil.");
+	
+	NSURL *url = [self urlForProductListingsWithParameters:@{ @"handle" : handle }];
+	return [self getRequestForURL:url completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		
+		BUYProduct *product = nil;
+		if (json && !error) {
+			product = [self.modelManager insertProductsWithJSONArray:json[BUYProductsKey]].firstObject;
+		}
+		
+		if (!product && !error) {
+			error = [self productsError];
+		}
+		block(product, error);
+	}];
+}
+
+- (BUYRequestOperation *)getProductById:(NSString *)productId completion:(BUYDataProductBlock)block
 {
 	BUYAssert(productId, @"Failed to get product by ID. Product ID must not be nil.");
 	
 	return [self getProductsByIds:@[productId] completion:^(NSArray *products, NSError *error) {
-		if (products.count > 0) {
-			block(products[0], error);
-		} else {
-			if (!error) {
-				error = [NSError errorWithDomain:BUYShopifyErrorDomain code:BUYShopifyError_InvalidProductID userInfo:@{ NSLocalizedDescriptionKey : @"Product ID is not valid. Confirm the product ID on your shop's admin and also ensure that the visibility is on for the Mobile App channel." }];
-			}
-			block(nil, error);
-		}
+		block(products.firstObject, error);
```

Comment:
It's used internally by `getProductsByIds:completion:` so the `error` in this case will never be `nil`.